recipes/xmlsec/all/conanfile.py

```diff
@@ -0,0 +1,213 @@
+from conans import ConanFile, tools, AutoToolsBuildEnvironment, VisualStudioBuildEnvironment
+from conans.errors import ConanInvalidConfiguration
+from contextlib import contextmanager
+import os
+
+class XmlSecConan(ConanFile):
+    name = "xmlsec"
+    description = "XML Security Library is a C library based on LibXML2. The library supports major XML security standards."
+    license = ("MIT", "MPL-1.1")
+    homepage = "https://github.com/lsh123/xmlsec"
+    url = "https://github.com/conan-io/conan-center-index"
+    generators = "pkg_config"
+    settings = "os", "compiler", "arch", "build_type"
+    topics = ("xml", "signature", "encryption")
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_openssl": [True, False],
+        "with_xslt": [True, False],
+        }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_openssl": True,
+        "with_xslt": False
+        }
+
+    _autotools = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def requirements(self):
+        self.requires("libxml2/2.9.10")
+        if self.options.with_openssl:
+            self.requires("openssl/1.1.1h")
+        if self.options.with_xslt:
+            self.requires("libxslt/1.1.34")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+            if self.settings.os == "Windows" and not self._is_msvc:
+                raise ConanInvalidConfiguration("Shared mingw builds are not supported (due to libtool blocking static dependencies)")
+        if not self.options.with_openssl:
+            raise ConanInvalidConfiguration("At least one crypto engine needs to be enabled")
```

Comment:
If False is not allowed, just remove this option?

Reply:
Other crypto engines can be added (nss, gnutls, ...).
It's just about future proofing.

Reply:
so I would expose another recipe's interface if only one crypto engine is allowed, something like this:

`crypto: ["openssl"]` # TODO: add other crypto engines

Reply:
I agree, maybe you'll get deprecated options. (e.g. libcurl)
@madebr Should I/we ;) change it?

Reply:
I also thought about this, but xmlsec allows multiple crypto backends.

Reply:
So if multiple crypto backend are allowed (are you sure? sometimes the real mechanism is hidden by build system options design), I'm ok with this option.

Reply:
See https://www.aleksey.com/xmlsec/faq.html
Every crypto backend creates a new `libxmlsec-XXXXX.so` library.

In practice a consumer will want to use only one crypto backend, but this package allows it.

Reply:
ok, then @steinerthomas, I'm fine with this option.